filers/default/src/main/java/org/commonjava/indy/filer/def/TimingOutputStream.java

```diff
@@ -0,0 +1,65 @@
+package org.commonjava.indy.filer.def;
+
+import org.commonjava.cdi.util.weft.ThreadContext;
+import org.commonjava.indy.metrics.RequestContextHelper;
+
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.TimeUnit;
+
+public class TimingOutputStream
+        extends FilterOutputStream
+{
+    private long nanos = -1L;
+
+    public TimingOutputStream( final OutputStream stream )
+    {
+        super( stream );
+    }
+
+    @Override
+    public void write( final int b )
+            throws IOException
+    {
+        if ( nanos < 0 )
```

Comment:
Java doc mentioned: The value returned by nanoTime() represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative). So it is not safe to use -1L. A boolean flag will be better.

Reply:
@ruhan1 ok, I switched it to a Long with an implied null initial value.